apps/aecontract/src/aect_contracts.erl

```diff
@@ -294,42 +330,100 @@ serialize_for_client(#contract{id           = Id,
 
 -spec deserialize(pubkey(), serialized()) -> contract().
 deserialize(Pubkey, Bin) ->
+    {?CONTRACT_TYPE, Vsn, RawFields} = aeser_chain_objects:deserialize_type_and_vsn(Bin),
+    deserialize(Pubkey, Vsn, RawFields).
+
+deserialize(Pubkey, ?PRE_IRIS_VSN, RawFields) ->
```

Comment:
I'm not fully sure that we store the contracts in serialized form in the DB

Reply:
We store in serialized AFAIK. How else would we store it? See `aect_state_tree:insert_contract`